src/lib/utils/translations.ts

```diff
@@ -15,3 +15,33 @@ export const isLangRightToLeft = (lang: Lang): boolean => {
 // Context: https://github.com/ethereum/ethereum-org-website/pull/5490#pullrequestreview-892596553
 export const getLocaleForNumberFormat = (locale: Lang): Lang =>
   locale === "fa" ? DEFAULT_LOCALE : locale
+
+export const getRequiredNamespacesForPath = (path: string) => {
+  let requiredNamespaces: string[] = ["common"]
+
+  if (path.startsWith("/community")) {
+    requiredNamespaces = [...requiredNamespaces, "page-community"]
+  }
+
+  if (path.startsWith("/energy-consumption")) {
+    requiredNamespaces = [
+      ...requiredNamespaces,
+      "page-about",
+      "page-what-is-ethereum",
+    ]
+  }
+
+  if (path.startsWith("/history")) {
+    requiredNamespaces = [...requiredNamespaces, "page-history"]
+  }
+
+  if (path.startsWith("/nft")) {
+    requiredNamespaces = [...requiredNamespaces, "learn-quizzes"]
+  }
+
+  if (path.startsWith("/staking")) {
+    requiredNamespaces = [...requiredNamespaces, "page-staking"]
+  }
+
```

Comment:
@pettinarip 
Looks good, curious if there was a preference for `let` then re-spreading, vs `const` and `.push()` here: 
ie. 
```tsx
const requiredNamespaces: string[] = ["common"]
// ...
if (criteria) { requiredNamespaces.push("namespace") }
```

Reply:
No special reason. I think it is just a matter of preference. For example, when you need to push more than just one element to the array, it is kind of annoying to do a double push 🤷🏼 and I'm more used to code with immutable methods.

Reply:
Same, I usually prefer methods that dont mutate the array like the spread op